### PR TITLE
added add_member_to_rank with sudo origin

### LINKED ISSUE #12029: Gov V2 Follow Up Issues
* Integrate a "fellowship" like setup into the kitchen sink runtime included in Substrate. This would have caught many of the issues with benchmarking which were debugged in: https://github.com/paritytech/substrate/pull/12022, https://github.com/paritytech/substrate/pull/12026, and https://github.com/paritytech/substrate/pull/12026
* Expose `add_member_to_rank` extrinsic which is callable by root to help bootstrap fellowship.

### PR DESCRIPTION
**Pull request by MrishoLukamba**
Fixes #12029 
Polkadot address: 133WyzJ4cEEbQ4twhT3nQpoXYeTCzeKn8kWfRfEi4XDF7wKh

### CODE REVIEW

**Comment by MrishoLukamba** [comment]
Solving issue #12029

**Comment by ggwpez** [comment]
So there are two Merge Requests now for the exposing of `add_member_to_rank`?  
You could just have fixed the other one, but anyway. Please close the out-dated one so we dont get confused.

**Comment by MrishoLukamba** [comment]
Done closing the other ones

File: frame/ranked-collective/src/lib.rs [hunk 1]
```diff
@@ -644,6 +644,17 @@ pub mod pallet {
 				pays_fee: Pays::No,
 			})
 		}
+
+		#[pallet::weight(100)]
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/lib.rs:648]: You need to write benchmarks and add the appropriate generated weights here

File: frame/ranked-collective/src/tests.rs [hunk 2]
```diff
@@ -308,6 +308,15 @@ fn promote_demote_works() {
 	});
 }
 
+#[test]
+pub fn add_member_to_rank_work(){
+	new_test_ext().execute_with(||{
+		assert_ok!(Club::add_member_to_rank(Origin::root(), 2,1));
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/tests.rs:314]: You should be checking the state is updated correctly for the member you are adding to rank

File: frame/ranked-collective/src/lib.rs [hunk 3]
```diff
@@ -644,6 +644,17 @@ pub mod pallet {
...
+		pub fn add_member_to_rank(
+			origin: OriginFor<T>,
+			member: T::AccountId,
+			rank: Rank
+		) -> DispatchResult {
+			let _= ensure_root(origin)?;
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/lib.rs:654]:
```suggestion
			ensure_root(origin)?;
```

File: frame/ranked-collective/src/lib.rs [hunk 4]
```diff
@@ -644,6 +644,17 @@ pub mod pallet {
...
+			Self::do_add_member_to_rank(member,rank)?;
+			Ok(())
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/lib.rs:656]:
```suggestion
			Self::do_add_member_to_rank(member,rank)
```

File: frame/ranked-collective/src/lib.rs [hunk 5]
```diff
@@ -644,6 +644,17 @@ pub mod pallet {
...
+
+		#[pallet::weight(T::WeightInfo::add_member_to_rank(*rank as u32))]
+		pub fn add_member_to_rank(
+			origin: OriginFor<T>,
+			member: T::AccountId,
+			rank: Rank
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			Self::do_add_member_to_rank(member,rank);
+			Ok(())
```
Comment (shawntabrizi) [inline frame/ranked-collective/src/lib.rs:656]:
```suggestion
			Self::do_add_member_to_rank(member,rank)
```

**Comment by shawntabrizi** [comment]
bot rebase

**Comment by MrishoLukamba** [comment]
> bot rebase

@Ank4n @kianenigma

**Comment by ggwpez** [comment]
bot rebase

**Comment by MrishoLukamba** [comment]
The PR is complete waiting for review only @ggwpez @bkchr

**Comment by ggwpez** [comment]
Code looks good. Please make the CI green.  
The main changes for rebasing: `Origin` renamed to `RuntimeOrigin` and `Weight` should be `u64` now in the weight file.

**Comment by ggwpez** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2103322

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2103321

**Comment by bkchr** [comment]
@MrishoLukamba if you want this to be merged, please make CI happy. Otherwise please close.

**Pull request closed without merging**